src/ServiceResetError.c

```diff
@@ -84,7 +84,15 @@ void Ros_ServiceResetError_Trigger(const void* request_msg, void* response_msg)
     }
 
     // Check for condition that can be fixed remotely
-    if (Ros_Controller_IsAlarm())
+    if (g_Ros_Controller.ioStatus[IO_ROBOTSTATUS_ALARM_MAJOR] != 0)
+    {
+        rosidl_runtime_c__String__assign(&response->message, "Major alarm active. Cannot be reset. Check teach pendant");
+        response->result_code.value = MOTION_NOT_READY_MAJOR_ALARM;
+        goto DONE;
+    } 
+    else if ((g_Ros_Controller.ioStatus[IO_ROBOTSTATUS_ALARM_MINOR] != 0)
+        ||  (g_Ros_Controller.ioStatus[IO_ROBOTSTATUS_ALARM_SYSTEM] != 0)
```

Comment:
@jimmy-mcelwain, were you able to confirm that `IO_ROBOTSTATUS_ALARM_SYSTEM` is resettable?

Reply:
Yes, system alarms are in the 9XXX range and can be reset. What we were talking about before are Off-line alarms, which don't have a dedicated I/O bit, but looking at documentation it looks like Off-line alarms are a subset of major alarms/the major alarm bit should be 1 if there is an offline alarm.